### ISSUE TITLE
Add gateway validation to VirtualServices

### ISSUE THREAD

**Issue by xeviknal** [kiali/kiali#2932]
Right now, kiali only validate gateways in the spec.gateways field. There is no analysis for each route.gateways item. See the following example:

```
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: orahub-vs
  namespace: istio-trials
spec:
  hosts:
    - orahub.oci.oraclecorp.com
  gateways:
    - my-egress-gw
    - mesh
  http:
    - match:
        - gateways:
            - mesh
          port: 80
      route:
        - destination:
            host: istio-egressgateway.istio-system.svc.cluster.local
            port:
              number: 80
    - match:
        - gateways:
            - my-egress-gw
          port: 80
      route:
        - destination:
            subset: tls-origination
            host: orahub.oci.oraclecorp.com
            port:
              number: 443
```

`my-egress-gw` gateway should trigger a validation when there isn't any gateway available with this name.

**Comment by lucasponce** [comment]
@xeviknal to me sounds a valid use case, I thought this case was already implemented.

**Comment by xeviknal** [comment]
It was implemented for the main gateway list, not for each route. It should be pretty straightforward to implement that.

**Comment by lucasponce** [comment]
Added to the Sprint to review priorities.

**Comment by lucasponce** [comment]
Closed by #4158 

@leandroberetta you can use the link syntax [1] in the PRs, then when the work is merged the issue will be closed automatically :-)

[1] https://docs.github.com/es/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
